internal/services/environment_settings/datasource_environment_settings_test.go

```diff
@@ -40,6 +40,7 @@ func TestAccTestEnvironmentSettingsDataSource_Validate_Read(t *testing.T) {
 					resource.TestCheckResourceAttr("data.powerplatform_environment_settings.settings", "audit_and_logs.audit_settings.is_audit_enabled", "false"),
 					resource.TestCheckResourceAttr("data.powerplatform_environment_settings.settings", "audit_and_logs.audit_settings.is_read_audit_enabled", "false"),
 					resource.TestCheckResourceAttr("data.powerplatform_environment_settings.settings", "audit_and_logs.audit_settings.is_user_access_audit_enabled", "false"),
+					resource.TestCheckResourceAttr("data.powerplatform_environment_settings.settings", "audit_and_logs.audit_settings.log_retention_period_in_days", "null"),
```

Comment:
if its empty it shoudn't be null, but the value should not exist on the resource as its not managed by it. You should validate use `resource.TestCheckModuleNoResourceAttr()`

Reply:
Resolved online.